pages/api/v1/applicants/[id].ts

```diff
@@ -1,19 +1,50 @@
-import { NextApiRequest, NextApiResponse } from 'next';
-import { EXAMPLE_USER } from '../../../../common/constants';
-import { protect } from '../../../../server/protect';
+import { string } from 'joi';
+import { ObjectId } from 'mongodb';
+import { NextApiHandler } from 'next';
+import { SingleApplicantApiResponse } from '../../../../common/types';
+import { connectToDatabase } from '../../../../server/mongoDB';
+import { isAdmin, protect } from '../../../../server/protect';
 
-export default protect(function handler(
-  req: NextApiRequest,
-  res: NextApiResponse
-) {
+const applicantHandler: NextApiHandler = async (req, res) => {
   switch (req.method) {
     // get a single applicant
     case 'GET':
-      return res.status(200).json(EXAMPLE_USER);
+      await getApplicant(req, res);
+      break;
     // update a single applicant
     case 'POST':
-      return res.status(201).send(undefined);
+      await postApplicant(req, res);
+      break;
     default:
       return res.status(405).setHeader('Allow', 'GET, POST').send(undefined);
   }
-});
+};
+
+const getApplicant: NextApiHandler = async (req, res) => {
+  const admin = await isAdmin(req);
+  if (!admin) {
+    return res.status(401).send({ message: 'User is not an admin' });
+  }
```

Comment:
pull this check out of the get/post handlers and just put it once in `applicantHandler`, before the switch case